modules/xfmg/oas/sharedlib/xyna-openapi/generator-utils/src/main/java/com/gip/xyna/openapi/codegen/XynaCodegenProperty.java

```diff
@@ -140,12 +150,12 @@ private String capitalize(String in) {
    * of primitive types
    */
   private boolean isPrimitive(CodegenPropertyInfo property) {
-    return property.getIsPrimitiveType() || property.getIsEnumOrRef() || property.getComplexType() == null
+    return !"object".equalsIgnoreCase(property.getDataType()) &&
```

Comment:
Do we only need to check for object? I have expected something like 
return !isGenericJsonObject && ...